src/core/config/browsers/firefox-helper.ts

```diff
@@ -0,0 +1,115 @@
+import { DebugConfiguration } from 'vscode';
+
+import isDocker from 'is-docker';
+import { CustomLauncher } from 'karma';
+
+import { GeneralConfigSetting, ProjectConfigSetting } from '../config-setting.js';
+import { ConfigStore } from '../config-store.js';
+import { ContainerMode } from '../extension-config.js';
+import { BrowserHelper } from './browser-helper.js';
+
+export class FirefoxBrowserHelper implements BrowserHelper {
+  public static DEFAULT_DEBUGGING_PORT: number | undefined = 9222;
+  private static HEADLESS_FLAGS: string[] = ['-headless'];
+
+  public get supportedBrowsers(): string[] {
+    return ['Firefox'];
+  }
+  public get debuggerType(): string {
+    return 'firefox';
+  }
+  public get debuggingPortFlag(): string {
+    return '-start-debugger-server';
+  }
+
+  public getCustomLauncher(
+    browserType: string,
+    customLaucher: CustomLauncher | undefined,
+    config: ConfigStore<ProjectConfigSetting>
+  ): CustomLauncher {
+    const configuredLauncher: CustomLauncher =
+      customLaucher ??
+      ({
+        base: browserType,
+        flags: [
+          ...FirefoxBrowserHelper.HEADLESS_FLAGS,
+          `${this.debuggingPortFlag} ${FirefoxBrowserHelper.DEFAULT_DEBUGGING_PORT}`
+        ],
+        prefs: {
+          'devtools.debugger.remote-enabled': true,
+          'devtools.chrome.enabled': true,
+          'devtools.debugger.prompt-connection': false
+        }
+      } as any);
+
+    if (!this.isSupportedBrowser(configuredLauncher.base)) {
+      return configuredLauncher;
```

Comment:
Yes, if you were to do that, that would happen. The theory, though, is that if you had `browserType = 'Chrome'` the `BrowserFactory` would return the `ChromeHelper` rather than the `FirefoxHelper`. Same as the comment you put on the `ElectronHelper`. This check is more for the case where `browserType = 'Firefox'` but the `customLauncher.base` value is somehow something entirely different.